process/block/alteredaccounts/alteredAccountsProvider.go

```diff
@@ -37,57 +38,46 @@ type ArgsAlteredAccountsProvider struct {
 }
 
 type alteredAccountsProvider struct {
-	shardCoordinator            sharding.Coordinator
-	addressConverter            core.PubkeyConverter
-	accountsDB                  state.AccountsAdapter
-	marshalizer                 marshal.Marshalizer
-	fungibleTokensIdentifier    map[string]struct{}
-	nonFungibleTokensIdentifier map[string]struct{}
+	shardCoordinator   sharding.Coordinator
+	addressConverter   core.PubkeyConverter
+	accountsDB         state.AccountsAdapter
+	marshalizer        marshal.Marshalizer
+	tokensProc         *tokensProcessor
+	mutExtractAccounts sync.Mutex
 }
 
 // NewAlteredAccountsProvider returns a new instance of alteredAccountsProvider
 func NewAlteredAccountsProvider(args ArgsAlteredAccountsProvider) (*alteredAccountsProvider, error) {
 	if check.IfNil(args.ShardCoordinator) {
-		return nil, ErrNilShardCoordinator
+		return nil, errNilShardCoordinator
 	}
 	if check.IfNil(args.AddressConverter) {
-		return nil, ErrNilPubKeyConverter
+		return nil, errNilPubKeyConverter
 	}
 	if check.IfNil(args.AccountsDB) {
-		return nil, ErrNilAccountsDB
+		return nil, errNilAccountsDB
 	}
 	if check.IfNil(args.Marshalizer) {
-		return nil, ErrNilMarshalizer
+		return nil, errNilMarshalizer
 	}
 
 	return &alteredAccountsProvider{
 		shardCoordinator: args.ShardCoordinator,
 		addressConverter: args.AddressConverter,
 		accountsDB:       args.AccountsDB,
 		marshalizer:      args.Marshalizer,
-		fungibleTokensIdentifier: map[string]struct{}{
-			core.BuiltInFunctionESDTTransfer:         {},
-			core.BuiltInFunctionESDTBurn:             {},
-			core.BuiltInFunctionESDTLocalMint:        {},
-			core.BuiltInFunctionESDTLocalBurn:        {},
-			core.BuiltInFunctionESDTWipe:             {},
-			core.BuiltInFunctionMultiESDTNFTTransfer: {},
-		},
-		nonFungibleTokensIdentifier: map[string]struct{}{
-			core.BuiltInFunctionESDTNFTTransfer:      {},
-			core.BuiltInFunctionESDTNFTBurn:          {},
-			core.BuiltInFunctionESDTNFTAddQuantity:   {},
-			core.BuiltInFunctionESDTNFTCreate:        {},
-			core.BuiltInFunctionMultiESDTNFTTransfer: {},
-		},
+		tokensProc:       newTokensProcessor(args.ShardCoordinator),
```

Comment:
To be discussed if it is the best idea to internally initialize a processor inside your altered accounts provider.
The major disadvantage  is that you cannot mock it for tests

Reply:
I just split the functionality into multiple components so the code is easier to track. In this situation, I am not interested in mocking the tokens processor, but on the contrary, I want to test the component like in production (so the real tokens processor)

Reply:
Agree, but this makes it easy to only test "happy-paths". If you want to "inject" an error to check how different code paths are executed, it would be harder, but still doable. Can leave it as it is, though.